src/renderer/services/midgard/utils.ts

```diff
@@ -142,3 +151,11 @@ export const toPoolData = (detail: PoolDetail): PoolData => ({
 export const filterPoolAssets = (poolAssets: string[]) => {
   return poolAssets.filter((poolAsset) => !isMiniToken(assetFromString(poolAsset) || { symbol: '' }))
 }
+
+export const getPoolAddressByChain = (endpoints: ThorchainEndpoints, chain: Chain): O.Option<PoolAddress> =>
```

Comment:
in general seems we will always (maybe im wrong?) have chain info from `Option<Asset>` so maybe we could encapsulate logic of getting `chain` here ?
```ts
  FP.pipe(
    oAsset,
    O.map(({ chain }) => chain),
    O.chain((chain) =>
      FP.pipe(
        endpoints,
        A.findFirstMap((endpoint) => (endpoint.chain === chain ? O.fromNullable(endpoint.address) : O.none))
      )
    ),
  )
```

Reply:
Don't agree. We might have a case where a knowledge about `Chain` (but not `Asset`) is available only.